ChatLayout/Classes/Core/Model/ItemModel.swift

```diff
@@ -67,7 +69,7 @@ struct ItemModel {
     // We are just resetting `calculatedSize` if needed as the actual size will be found in
     // `invalidationContext(forPreferredLayoutAttributes:, withOriginalAttributes:)`.
     // It is important for the rotation to keep previous frame size.
-    mutating func resetSize() {
+    nonisolated mutating func resetSize() {
```

Comment:
Regarding things I do not understand wrt isolation and Swift, I don’t really understand how it is possible here that you access and set the `calculatedSize` and `preferredSize` while you are non-isolated. It’s weird.
My guess is, because the function is `mutating` we’re working in a _copy_ of the struct. A copy which is guaranteed to be uniquely referenced, so changing the memory, as long as you don’t access an `@MainActor` function, is ok.
But I’m really not sure.
Would be interesting to compile this with an older version of the compiler (~5.9 or 5.8, I guess) see if the code still compiles. I know Apple added a **lot** of rules to simplify the transition to strict concurrency check, that removed some unneeded restrictions. Maybe it’s one of them.

Reply:
Yeah I think my theory makes sense; I tried accessing `size`, which is a _computed_ property, and I’m told it’s not possible because I’m not `@MainActor` in the function.